packages/PHPStanStaticTypeMapper/TypeMapper/UnionTypeMapper.php

```diff
@@ -123,11 +123,21 @@ public function mapToPhpParserNode(Type $type, string $typeKind): ?Node
             return $this->matchTypeForUnionedObjectTypes($type, $typeKind);
         }
 
+        return $this->mapNullabledType($nullabledType, $typeKind);
+    }
+
+    private function mapNullabledType(Type $nullabledType, string $typeKind): ?Node
+    {
         // void cannot be nullable
         if ($nullabledType instanceof VoidType) {
             return null;
         }
 
+        // null is already part of mixed
+        if ($nullabledType instanceof MixedType) {
+            $nullabledType = new MixedType();
+        }
```

Comment:
my original suggestion at https://github.com/rectorphp/rector/issues/7501#issuecomment-1254730274 is the correct way for it, if the type is `MixedType`, the tranfsormed parsed node can be `null` or `Name('mixed')` based on php version feature enabled.

so, this should be removed:

```suggestion
```

and add check : 

```diff
-        if (! $this->nodeNameResolver->isName($nullabledTypeNode, 'false')) {
+        if (! $this->nodeNameResolver->isNames($nullabledTypeNode, ['false', 'mixed'])) {
```

in line 151 before apply Nullable into parsed Node.

Reply:
In that case I think we lose information in newer PHP versions that do support `mixed`, right? Something that could have been `mixed` will have no type instead. That is similar to how the `mapToPhpParserNode` method changes `true|false` to `bool`, since they are equivalent. Similarly, `null|mixed` is equivalent to `mixed`.

Reply:
That's different possible feature to implement, that require failing test case to avoid regression apply `mixed` everywhere, this PR should solve only the issue related.

Reply:
Ok, clear. I will go with your original suggestion and finish up the PR. Thanks for the help!